packages/web3-core-method/src/index.js

```diff
@@ -547,22 +548,35 @@ Method.prototype._confirmTransaction = function (defer, result, payload) {
 
     // start watching for confirmation depending on the support features of the provider
     var startWatching = function (existingReceipt) {
+        let blockHeaderArrived = false; 
+
         const startInterval = () => {
             intervalId = setInterval(checkConfirmation.bind(null, existingReceipt, true), 1000);
         };
 
-        if (!this.requestManager.provider.on) {
-            startInterval();
-        } else {
-            _ethereumCall.subscribe('newBlockHeaders', function (err, blockHeader, sub) {
-                if (err || !blockHeader) {
-                    // fall back to polling
-                    startInterval();
-                } else {
-                    checkConfirmation(existingReceipt, false, err, blockHeader, sub);
-                }
-            });
+        // If provider do not support event subscription use polling
+        if(!this.requestManager.provider.on) {
```

Comment:
Shouldnt we look specific event subscription support  in provider that we are interested in `newBlockHeaders` ,

Reply:
There is no way to check if a provider a particular event or not.